sonic_platform_base/chassis_base.py

```diff
@@ -95,16 +98,39 @@ def get_reboot_cause(self):
         """
         raise NotImplementedError
 
-    def get_component_versions(self):
+    def get_component_list(self):
+        """
+        Retrieves chassis components list such as BIOS, CPLD, FPGA, etc.
+
+        Returns:
+            A list containing component name.
+        """
+        return self._component_list
+
+    def get_firmware_version(self, component):
```

Comment:
`def get_firmware_version(self, component_name):`, again, to go along with my above comments?